<launch>
    <arg name="sim_time_required" default="true"/>
    <param name="use_sim_time" value="$(arg sim_time_required)"/>

    <arg name="robot_name" default="sparkal1"/>
    <arg name="robot_id" default="0"/>
    <arg name="use_gt_frame" default="false"/>
    <arg name="robot_frame" default="$(arg robot_name)/base" if="$(arg use_gt_frame)"/>
    <arg name="robot_frame" default="base_link" unless="$(arg use_gt_frame)"/>
    <arg name="odom_frame" default="$(arg robot_name)/odom"/>
    <arg name="world_frame" default="$(arg robot_name)/map"/>
    <arg name="use_compressed_transport" default="true"/>

    <arg name="labelspace_name" default="ade20k_mp3d"/>
    <arg name="semantic_map_dir" default="$(find semantic_recolor)/config/colors"/>
    <arg name="semantic_map_file" default="$(arg labelspace_name).csv"/>
    <arg name="semantic_map_path" default="$(arg semantic_map_dir)/$(arg semantic_map_file)"/>

    <arg name="pointcloud_topic" value="semantic_pointcloud"/>
    <arg name="rgb_info_topic" default="forward/color/camera_info"/>
    <arg name="rgb_topic" default="forward/color/image_raw"/>
    <arg name="depth_topic" default="forward/depth/image_rect_raw"/>
    <arg name="label_topic" default="semantic/image_raw"/>

    <include file="$(find hydra_ros)/launch/utils/data_sources/jackal_pointcloud.launch"
             pass_all_args="true"/>

    <include file="$(find hydra_ros)/launch/hydra.launch" ns="$(arg robot_name)" pass_all_args="true">
        <arg name="config_dir" default="$(find hydra)/config/simmons_jackal"/>
        <arg name="extrinsics_mode" value="USE_KIMERA"/>
        <arg name="kimera_extrinsics_file" default="$(find hydra_vio_configs)/config/robots/$(arg robot_name)/LeftCameraParams.yaml"/>
        <arg name="rviz_file" default="$(arg robot_name).rviz"/>
        <arg name="use_oriented_bounding_boxes" default="true"/>
        <arg name="rgb_image_transport" value="compressed" if="$(arg use_compressed_transport)"/>
        <arg name="rgb_image_transport" value="raw" unless="$(arg use_compressed_transport)"/>
        <arg name="depth_topic" default="depth_registered/image_rect"/>
    </include>

</launch>
